recipe/migrations/python314.yaml

```diff
@@ -6,7 +6,7 @@
 migrator_ts: 1724712514
 __migrator:
     commit_message: Rebuild for python 3.14
-    migration_number: 1
+    migration_number: 2
```

Comment:
That's because I dislike bumping `migration_number` - not even due to the extra PR churn, but because it becomes much more confusing for the average user why a given migration (e.g. with `migration_number: 1`) suddenly doesn't apply anymore. I think we should try to avoid this for such a big and long-running migration as 3.14 will be.